backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/user/api/impl/UserDAO.java

```diff
@@ -318,4 +326,48 @@ public void save(@NonNull final org.compiere.model.I_AD_User user)
 		return OrgMappingId.optionalOfRepoId(sourceUserRecord.getAD_Org_Mapping_ID());
 	}
 
+	@Override
+	public void deleteUserSubstituteByUserId(final UserId userId)
+	{
+		queryBL.createQueryBuilder(I_AD_User_Substitute.class)
+				.addEqualsFilter(I_AD_User_Substitute.COLUMNNAME_AD_User_ID, userId)
+				.create()
+				.delete();
+	}
+
+	@Override
+	public void deleteUserMailByUserId(final UserId userId)
+	{
+		queryBL.createQueryBuilder(I_AD_UserMail.class)
+				.addEqualsFilter(I_AD_UserMail.COLUMNNAME_AD_User_ID, userId)
+				.create()
+				.delete();
+	}
+
+	@Override
+	public void deleteUserQueryByUserId(final UserId userId)
+	{
+		queryBL.createQueryBuilder(I_AD_UserQuery.class)
+				.addEqualsFilter(I_AD_UserQuery.COLUMNNAME_AD_User_ID, userId)
+				.create()
+				.delete();
+	}
+
+	@Override
+	public void deleteUserRolesByUserId(final UserId userId)
```

Comment:
pls, move it to RoleDAO.

Basically, each database table shall be handled in its own Repository and nowhere else!

Reply:
Done